HW4/HW4.py

```diff
@@ -0,0 +1,381 @@
+import sys
+
+import numpy
+import pandas as pd
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from plotly.subplots import make_subplots
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import confusion_matrix
+
+
+def main():
+    """Set the arguments below until I can get argparse in here"""
+    # arguments: df, response (string), predictors (list)
+    # testing data only
+    # names = ["Sepal Length", "Sepal Width", "Petal Length", "Petal Width", "Class"]
+    # df = pd.read_csv("iris.data", names=names)
+    # df['boo_response']=df['Class'].apply(lambda x: True if x=='Iris-setosa' else False)
+
+    # Hey user, set these until I get argparse in here
+    df = pd.DataFrame()
+    response = ""
+    predictors = []
+    # predictor = 'Sepal Length'
+
+    # testing only
+    # create boolean response - probably should remove
+    # df['boo_response']=df['Class'].apply(lambda x: True if x=='Iris-setosa' else False)
+
+    # steps
+    # Determine if response is continuous or boolean
+    y = response
+    if df[response].unique().size > 2:
+        response_type = "continuous"
+    else:
+        # hard coding to boolean rather than categorical
+        response_type = "boolean"
+
+    # Determine if predictors are categorical or continuous
+    cont_master_list = df._get_numeric_data().columns.to_list()
+    predictor_types = {}
+    cont_predictor_col_names = []
+    for predictor in predictors:
```

Comment:
maybe try writing a function to test if the columns is continuous or not. I wrote a simple one that I applied to everything but im not sure it actually catches all cases